packages/fetchai/skills/ml_train/tasks.py

```diff
@@ -35,46 +37,34 @@ def __init__(
         skill_context: SkillContext,
         train_data: Tuple[np.ndarray, np.ndarray],
         epochs_per_batch: int = 10,
-        batch_size: int = 32,
+        weights: Optional[int] = None,
     ):
         """Initialize the task."""
         super().__init__(logger=skill_context.logger)
-        self.train_x, self.train_y = train_data
 
-        self.model = self._make_model()
+        self.train_x, self.train_y = train_data
+        self.weights = weights
         self.epochs_per_batch = epochs_per_batch
-        self.batch_size = batch_size
+
+        self.ml_model = cast(MLModel, skill_context.ml_model)
 
     def setup(self) -> None:
         """Set up the task."""
         self.logger.info("ML Train task: setup method called.")
 
-    @staticmethod
-    def _make_model() -> Any:
-        """Make the model."""
-        import tensorflow as tf  # pylint: disable=import-outside-toplevel
-
-        model = tf.keras.Sequential(
-            [
-                tf.keras.layers.Flatten(input_shape=(28, 28)),
-                tf.keras.layers.Dense(128, activation="relu"),
-                tf.keras.layers.Dense(10, activation="softmax"),
-            ]
-        )
-        model.compile(
-            optimizer="adam",
-            loss="sparse_categorical_crossentropy",
-            metrics=["accuracy"],
-        )
-        return model
-
     def execute(self, *args: Any, **kwargs: Any) -> Any:
         """Execute the task."""
         self.logger.info("Start training with {} rows".format(self.train_x.shape[0]))
-        self.model.fit(self.train_x, self.train_y, epochs=self.epochs_per_batch)
-        loss, acc = self.model.evaluate(self.train_x, self.train_y, verbose=2)
+
+        model = self.ml_model.make_model(self.weights)
```

Comment:
1. make model from static method above
2. set weight on model from weight added in constructor
3. we call fit
4. get weights
5. print loss, acc
6. we return the new weights

Reply:
so only 1 needs to change